### PR TITLE
SonarCloud "Major" Bugs

### PR DESCRIPTION
**Pull request by johnmay**
Working through these - let me know any thoughts.

### CODE REVIEW

File: legacy/src/main/java/org/openscience/cdk/isomorphism/matchers/SymbolQueryAtom.java [hunk 1]
```diff
@@ -47,7 +49,7 @@ public void setHCount(int m_HCount) {
     @Override
     public boolean matches(IAtom atom) {
         if (ID != null && HCount == 0)
-            return this.getSymbol() != (atom.getSymbol());
+            return !Objects.equals(this.getSymbol(), atom.getSymbol());
```
Comment (egonw) [inline legacy/src/main/java/org/openscience/cdk/isomorphism/matchers/SymbolQueryAtom.java:52]: That's an actual bug fix...

File: storage/io/src/main/java/org/openscience/cdk/io/ShelXReader.java [hunk 2]
```diff
@@ -256,7 +256,6 @@ private ICrystal readCrystal(ICrystal crystal) throws IOException {
             } else if (command.equalsIgnoreCase("FRAG")) {
             } else if (command.equalsIgnoreCase("FEND")) {
             } else if (command.equalsIgnoreCase("EXYZ")) {
-            } else if (command.equalsIgnoreCase("EXTI")) {
```
Comment (egonw) [inline storage/io/src/main/java/org/openscience/cdk/io/ShelXReader.java:259]: Okay, was that really needed? :)

BTW, this was actually part of my PhD thesis... I really hope to find some time to go back to this code and make my PhD work more accessible from the CDK...

**Review by egonw — APPROVED** [review]
Yeah, looks good to me.

**Comment by johnmay** [comment]
Yeah it definitely does highlight bugs, like I said a lot of these are in corners of the CDK that are less frequently used

File: storage/io/src/main/java/org/openscience/cdk/io/ShelXReader.java (hunk 2 above)
Comment (johnmay) [reply storage/io/src/main/java/org/openscience/cdk/io/ShelXReader.java:259]: TBF I checked the manual and it really can't occur there :-).

Comment (egonw) [reply storage/io/src/main/java/org/openscience/cdk/io/ShelXReader.java:259]: Another bug.

**Pull request merged**